base/io.jl

```diff
@@ -563,12 +563,17 @@ function write(s::IO, a::SubArray{T,N,<:Array}) where {T,N}
 end
 
 function write(io::IO, c::Char)
-    u = bswap(reinterpret(UInt32, c))
-    n = 1
-    while true
-        write(io, u % UInt8)
-        (u >>= 8) == 0 && return n
-        n += 1
+    v = reinterpret(UInt32, c)
+    u = hton(v) # BIG-endian
+    p = unsafe_convert(Ptr{Cvoid}, Ref{UInt32}(u))
```

Comment:
I believe this Ref needs a `GC.@preserve`.

Reply:
Yes, it requires `GC.@preserve` for `Ref`.
```julia
function f0(c::Char)
    u = hton(reinterpret(UInt32, c))
    p = unsafe_convert(Ptr{Cvoid}, Ref{UInt32}(u))
    show(unsafe_wrap(Array{UInt8,1}, Ptr{UInt8}(p), (4,)))
end

function f1(c::Char)
    u = hton(reinterpret(UInt32, c))
    GC.@preserve u begin
        p = unsafe_convert(Ptr{Cvoid}, Ref{UInt32}(u))
        show(unsafe_wrap(Array{UInt8,1}, Ptr{UInt8}(p), (4,)))
    end
end

function f2(c::Char)
    u = Ref{UInt32}(hton(reinterpret(UInt32, c)))
    GC.@preserve u begin
        p = unsafe_convert(Ptr{Cvoid}, u)
        show(unsafe_wrap(Array{UInt8,1}, Ptr{UInt8}(p), (4,)))
    end
end
```

`UInt8[0xed, 0x95, 0x9c, 0x00]` must be printed:
```
julia> f0('한')
UInt8[0x40, 0x27, 0x3e, 0x3c]
julia> f0('한')
UInt8[0x01, 0x00, 0x00, 0x00]
julia>

julia> f1('한')
UInt8[0x20, 0x1c, 0x3e, 0x3c]
julia> f1('한')
UInt8[0x01, 0x00, 0x00, 0x00]
julia>

julia> f2('한')
UInt8[0xed, 0x95, 0x9c, 0x00]
julia> f2('한')
UInt8[0xed, 0x95, 0x9c, 0x00]
julia>
```

Reply:
Right – if `@time` isn't showing the extra allocation of the `Ref` object, this is broken (as you are demonstrating above)